core/src/test/scala/dagr/core/execsystem/TaskExecutionRunnerTest.scala

```diff
@@ -120,6 +120,7 @@ class TaskExecutionRunnerTest extends UnitSpec with OptionValues with BeforeAndA
     // run the task
     taskRunner.runTask(taskInfo = taskInfo)
     taskRunner.runningTaskIds should have size 1 // there should be a running task
+    taskRunner.joinAll()
```

Comment:
I still think it makes sense to have some kind of timeout, and fail if not all of them were joined, since I don't want tests that could run forever.